modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/HttpClient.java

```diff
@@ -32,57 +34,97 @@
 
 class HttpClient {
 
-    byte[] getBytes(String url) throws IOException {
-        return get(url).readAllBytes();
+    // a private sentinel value for representing the idea that there's no auth for some request.
+    // this allows us to have a not-null requirement on the methods that do accept an auth.
+    // if you don't want auth, then don't use those methods. ;)
+    private static final PasswordAuthentication NO_AUTH = new PasswordAuthentication("no_auth", "no_auth_unused".toCharArray());
+
+    PasswordAuthentication auth(final String username, final String password) {
+        return new PasswordAuthentication(username, password.toCharArray());
+    }
```

Comment:
> Or is the usage going to be extended in a follow-up PR?

No overwhelming reason -- in my calling code I thought it was slightly prettier to have it be instance rather than static.

Reply:
Alright, I don't feel too strongly about making it instance or static.

What about my other comment regarding the class that this method is in? I'm usually not a fan of putting test code in production code.